lib/src/command/cache_clean.dart

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import '../command.dart';
+import '../io.dart';
+import '../log.dart' as log;
+
+class CacheCleanCommand extends PubCommand {
+  @override
+  String get name => 'clean';
+  @override
+  String get description => 'Clears the entire system cache.';
+  @override
+  bool get takesArguments => false;
+
+  @override
+  Future<void> runProtected() async {
+    if (entryExists(cache.rootDir)) {
```

Comment:
I'm tempted to suggest the user should confirm this action.

Maybe we have:
 * `dart pub cache clean --force`, no prompt.
 * `dart pub cache clean`, prompts user on the CLI.

**Reasoning:** deleting the entire cache means that anything installed will stop working.
What about flutter? will it recover from this?

Reply:
Yeah - probably better do this!

Reply:
I changed the code, so the (now empty) directory will be created again. That will make sure flutter keeps using the internal cache if it is already doing so.